program/rust/src/c_oracle_header.rs

```diff
@@ -34,94 +34,137 @@ pub trait PythAccount: Pod {
 
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_acc {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_acc {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_map_table {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_map_table {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_prod {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_prod {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_price {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_price {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_price_info {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_price_info {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_ema {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_ema {
-}
-
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_pub_key_t {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_pub_key_t {
-}
-
-
-#[cfg(target_endian = "little")]
-unsafe impl Zeroable for pc_price_comp_t {
-}
-
-#[cfg(target_endian = "little")]
-unsafe impl Pod for pc_price_comp_t {
-}
-
-impl pc_pub_key_t {
-    pub fn new_unique() -> pc_pub_key_t {
+#[repr(C)]
+#[derive(Copy, Clone, Pod, Zeroable)]
+pub struct PriceAccount {
+    pub magic_:          u32,
+    pub ver_:            u32,
+    pub type_:           u32,
+    pub size_:           u32,
+    pub ptype_:          u32,      // Type of the price account
+    pub expo_:           i32,      // Exponent for the published prices
+    pub num_:            u32,      // Current number of authorized publishers
+    pub num_qt_:         u32,      // Number of valid quotes for the last aggregation
+    pub last_slot_:      u64,      // Last slot with a succesful aggregation (status : TRADING)
+    pub valid_slot_:     u64,      // Second to last slot where aggregation was attempted
+    pub twap_:           PriceEma, // Ema for price
+    pub twac_:           PriceEma, // Ema for confidence
+    pub timestamp_:      i64,      // Last time aggregation was attempted
+    pub min_pub_:        u8,       // Minimum valid publisher quotes for a succesful aggregation
+    pub unused_1_:       i8,
+    pub unused_2_:       i16,
+    pub unused_3_:       i32,
+    pub prod_:           CPubkey, // Corresponding mapping account
+    pub next_:           CPubkey, // Next price account in the list
+    pub prev_slot_:      u64,     /* Second to last slot where aggregation was succesful
+                                   * (status : TRADING) */
+    pub prev_price_:     i64,       // Aggregate price at prev_slot_
+    pub prev_conf_:      u64,       // Confidence interval at prev_slot_
+    pub prev_timestamp_: i64,       // Timestamp of prev_slot_
+    pub agg_:            PriceInfo, // Last attempted aggregate results
+    pub comp_:           [PriceComponent; 32usize], // Publisher's price components
```

Comment:
Nitpick: Doesn’t need fixing but array sizes are always `usize` so can be specified as just 32.

Reply:
In the future they'll be constants